New Readme file

CarolRameder opened the pull request:
Modified the Readme according to feedback

CarolRameder committed:
Update README.md

CarolRameder committed:
Update README.md

CarolRameder committed:
Added headers for each experiment section

CarolRameder committed:
Create empty

CarolRameder committed:
Uploaded metadata for multple images per patient

CarolRameder committed:
Deleted empty

CarolRameder committed:
UpdatedREADME.md

CarolRameder committed:
Update README.md

CarolRameder committed:
Delete Preprocess scripts/backup_heatmaps.ipynb

JasperO98 commented on README.md:
```diff
@@ -25,7 +25,7 @@ An [*anaconda*](https://www.anaconda.com/distribution/) environment is recommend
 
 
-The path value should be set to 'DATASET_DATA_DIR' as shown in [file](https://github.com/CarolRameder/NaroNet/blob/main/NaroNet-main/src/main.py).
+The path value should be set to 'DATASET_DATA_DIR' as shown in line 83 from [file](https://github.com/CarolRameder/NaroNet/blob/main/NaroNet-main/src/main.py).
```
Why do you call it DATASET_DATA_DIR path if there's not variable named like that? It's a bit confusing.
Can you also please add the code snippet by writing the code in between ```, see below.
```python
path = '/home/carol/NaroNet-main/NaroNet-main/Endometrial_POLE/'    
```

JasperO98 reviewed (changes requested):
I left a comment can you please resolve it thanks.

JasperO98 committed:
move code snippet of all steps

Move code snippet of all steps under the line that it's being referenced in, more clear.

JasperO98 commented on README.md:
```diff
@@ -68,14 +106,31 @@ conda env create -f idec.yaml
-Step 3: PCL inference process. With the trained weights, the model generates the embeddings. Run the main.py file again as in Step 2.
+Step 3: PCL inference process. With the trained weights, the model generates the embeddings. Rerun the main.py file as in Step 2.
 
 Step 4: Run the three cells from "Column-wise processing" section in the [file](https://github.com/CarolRameder/NaroNet/blob/main/EXP3/Pipeline.ipynb). This normalizes the embedding to the scale required by the Auto Encoder(AE) component. All the embeddings from all images in the dataset are merged in one ".npy". This is the input format for the AE.
```
can you add the code snippet here?

JasperO98 commented on README.md:
```diff
@@ -68,14 +106,31 @@ conda env create -f idec.yaml
+Step 3: PCL inference process. With the trained weights, the model generates the embeddings. Rerun the main.py file as in Step 2.
 
 Step 4: Run the three cells from "Column-wise processing" section in the [file](https://github.com/CarolRameder/NaroNet/blob/main/EXP3/Pipeline.ipynb). This normalizes the embedding to the scale required by the Auto Encoder(AE) component. All the embeddings from all images in the dataset are merged in one ".npy". This is the input format for the AE.
 
```
Can you add the code snippet for the "column-wise processing" and also for "laod_embs" in step 5. Also the one in step 7 when mentioning line 190. Please add a code snippet each time you reference a line of code

JasperO98 commented on README.md:
```diff
@@ -43,6 +41,10 @@ run_NaroNet(path,params)
 
+The path value should be set to 'DATASET_DATA_DIR' as shown in line 83 from [file](https://github.com/CarolRameder/NaroNet/blob/main/NaroNet-main/src/main.py).
+
+# Exp 1
```
Can you add a description of what you're trying to do in each experiment? You can just copy what you have from the thesis.

CarolRameder replied:
More details than the list at the start of the file? (Lines 5, 6, 7)

JasperO98 replied:
If you mention what you listed again in the headers it's fine

JasperO98 commented on README.md:
````diff
@@ -51,15 +53,51 @@ conda activate rerunnaro
+get_BioInsights(path,params)
+```
+
+# Exp 3
````
You haven't explained experiment 2?

CarolRameder replied:
Experiment 2 is completely similar to 1, besides the initial preprocessing steps. I can add a short sentence on this. We talked about skipping that from the public repo and I will give you the details on running the pipeline in another private file. Is that fine?

JasperO98 replied:
Yes please mention that exp2 is similar to 1. Why did we decide again on keeping it private?

CarolRameder replied:
The dataset used is in-house EMC and will not be made publicly available as far as I know.

JasperO98 replied:
I meant the methods you used not the dataset

JasperO98 replied:
The methods will be shared in a private file

JasperO98 commented on README.md:
```diff
@@ -134,9 +206,9 @@ DATASET_DATA_DIR/
 		
```
I think moving the preparing datasets before the experiments is more clear, because for some experiments you would have already needed to do this right?

JasperO98 replied:
What do you think?

CarolRameder replied:
"Preparing the datasets" section was moved before the experiments in the latest version, right?

JasperO98 commented on README.md:
````diff
@@ -162,6 +234,8 @@ Marker_4
+
 ## Preparing parameter configuration
 In the following sections (i.e., preprocessing, PCL, NaroNet, BioInsights) several parameters are required to be set. Although parameters will be explained in each section, all of them should be specified in the file named 'DatasetParameters.py', which is located in the folder 'NaroNet/src/utils'. Change it to your own configuration, where 'DATASET_DATA_DIR' is your target folder. See example [fi ...
 ```python
````
Here below you DATASET_DATA_DIR is shown in the code so is that what you meant in the beginning with DATASET_DATA_DIR?

CarolRameder replied:
Yes. Now it comes after the Dataset preparation section, so it should be easier to understand.

JasperO98 replied:
But the dataset preperation is still below the experiments section, right?

CarolRameder replied:
In the Readme file from the main branch, it was switched. Can you see it? That is the latest version.

JasperO98 replied:
Sorry yes i see now

CarolRameder committed:
Integrated feedback

CarolRameder committed:
Integrated last pieces of feedback.

JasperO98 commented on README.md:
Can you add a code snippet to step 8 in exp3.
Also, can you add the bash command you'd perform in step 9? How did you replace the initial embeddings with the new embeddings? Just remove the content of the folder add copying the new embeddings to the folder? Make this clear using code snippet of the bash command.

CarolRameder replied:
I did Step 9 manually. I will add this and Step 8.

CarolRameder committed:
Updated Steps 8 and 9

JasperO98 committed:
Convert manual steps into bash code
